core/src/main/java/org/jruby/util/invoke/MethodHandle.java

```diff
@@ -0,0 +1,71 @@
+package org.jruby.util.invoke;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class MethodHandle {
```

Comment:
This part is pretty nice, and I almost wonder if we should pull it out into its own library, for others to use.